### PR TITLE
add get ticket, post ticket, and path ticket/tickedId endpoints

### PR DESCRIPTION
**Pull request by krzysztofMi**
Endpoints:

**GET: /ticket?all=**
all = true returns all tickets if admin, all = false returns all current user's tickets

**POST: /ticket**
creating new tickets as current user. Json argument for this endpoint is ticket content

**PATH: /ticket/{tickedId**}
updating ticket if it is current user ticket or when logged user account type is admin

### CODE REVIEW

**Comment by kotcrab** [comment]
Please run code formatter on your code. You can see the build is failing due to linter checks:
http://149.156.146.249:60001/jenkins/blue/organizations/jenkins/backend/detail/PR-20/1/pipeline

File: api/src/main/kotlin/pl/edu/pk/backend/app/Routes.kt [hunk 1]
```diff
@@ -34,6 +34,12 @@ fun createRouter(vertx: Vertx, controllers: Controllers): Router {
     router.patch("/user/:email").handler(::patchUser)
   }
 
+  with(controllers.ticketController) {
+    router.get("/ticket").handler(::getTickets)
+    router.post("/ticket").handler(::postTicket)
+    router.patch("/ticket/:ticketId").handler(::pathTicket)
```
Comment (kotcrab) [inline api/src/main/kotlin/pl/edu/pk/backend/app/Routes.kt]: Typo `patchTicket`?

File: api/src/main/kotlin/pl/edu/pk/backend/controller/TicketController.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import pl.edu.pk.backend.util.ApiError
+
+class TicketController(private val ticketService: TicketService) {
+  fun getTickets(ctx: RoutingContext) {
+    if (!ctx.checkCurrentUserHasRole(Role.User)) return
+    var all = "false"
+    try {
+      all = ctx.queryParam("all")[0]
+    } catch (ex: Exception) {
+    }
```
Comment (kotcrab) [inline api/src/main/kotlin/pl/edu/pk/backend/controller/TicketController.kt]: Don't leave empty catch blocks and avoid using exceptions for normal control flow. 
You can rewrite this using:
```
ctx.queryParam("all").firstOrNull()?.toBoolean()
```

File: api/src/main/kotlin/pl/edu/pk/backend/controller/TicketController.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    if (all == "true") {
+      if (!ctx.checkCurrentUserHasRole(Role.Admin)) return
```
Comment (kotcrab) [inline api/src/main/kotlin/pl/edu/pk/backend/controller/TicketController.kt]: I recommend avoid using one line ifs.

File: core/src/main/kotlin/pl/edu/pk/backend/model/Ticket.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  val author: String,
+  val createData: String
+) {
+  companion object {
+    fun from(ticket: Ticket): TicketDto {
+      val formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy")
+      return TicketDto(
+        ticket.id,
+        ticket.closed,
+        "user_" + ticket.author.id,
```
Comment (kotcrab) [inline core/src/main/kotlin/pl/edu/pk/backend/model/Ticket.kt]: Use email here. I'll update API doc but every field with `user-1` etc. should be an email.

File: core/src/main/kotlin/pl/edu/pk/backend/service/TicketService.kt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import io.vertx.core.Future
+import io.vertx.core.json.JsonObject
+import pl.edu.pk.backend.model.TicketDto
+import pl.edu.pk.backend.repository.TicketRepository
+import pl.edu.pk.backend.repository.UserRepository
+import pl.edu.pk.backend.util.ValidationException
+
+class TicketService(private val ticketRepository: TicketRepository, private val userRepository: UserRepository) {
+  fun getTickets(): Future<List<TicketDto>> {
+    return ticketRepository.getAllTickets().map { it.map { TicketDto.from(it) } }
```
Comment (kotcrab) [inline core/src/main/kotlin/pl/edu/pk/backend/service/TicketService.kt]: Prefer to put map/compose on new line.

**Review by kotcrab — CHANGES_REQUESTED** [review]
Looks good! Just few comments. Please also update `api.yaml` with new endpoints.

https://github.com/projekt-zespolowy-lab04-2020/backend/blob/master/api/src/main/resources/webroot/api.yaml

Documentation:
https://swagger.io/docs/specification/basic-structure/

Online editor:
https://editor.swagger.io/

**Comment by kotcrab** [comment]
Also don't worry about failing CI check `default — Build finished. ` that was a misconfiguration. Should be fixed now but will probably be applied after new commit or new PR.

File: api/src/main/kotlin/pl/edu/pk/backend/app/Routes.kt (hunk 1 above)
Comment (krzysztofMi) [reply api/src/main/kotlin/pl/edu/pk/backend/app/Routes.kt]: I thought that is path ;D

**Review by kotcrab — APPROVED** [review]

**Comment by kotcrab** [comment]
Okay, you can merge this 👍

**Comment by kotcrab** [comment]
I fixed few issues with api.yml, please validate in the online editor next time, looks good otherwise. You can start working on part two of #16 .

**Pull request merged**